frontend/src/scenes/insights/insightSceneLogic.tsx

```diff
@@ -130,6 +132,20 @@ export const insightSceneLogic = kea<insightSceneLogicType>({
             const insightId = String(shortId) as InsightShortId
             const oldInsightId = values.insightId
             if (insightId !== oldInsightId || insightMode !== values.insightMode) {
+                // If navigating from an unsaved insight to a different insight within the scene, prompt the user
+                if (
+                    sceneLogic.values.scene === Scene.Insight &&
+                    insightId !== oldInsightId &&
+                    oldInsightId !== 'new' &&
+                    values.insightCache?.logic.values.filtersChanged
+                ) {
+                    if (confirm('Leave insight? Changes you made may not be saved.')) {
+                        values.insightCache.logic.actions.discardChanges()
+                    } else {
+                        history.back()
+                        return
+                    }
+                }
```

Comment:
The code here and in `sceneLogic` seems like the most sensible way to do this (lesser of all evils), but could both nearly identical codepaths (here and  `sceneLogic`) be merged somehow? It might even be possible to just run this only in the `openScene` block, as that action should be called as well when the `id` in the url changes.